src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -220,7 +237,7 @@ internal static string TryExpandShortFileName(ref ValueStringBuilder outputBuild
             // Strip out any added characters at the front of the string
             ReadOnlySpan<char> output = builderToUse.AsSpan(rootDifference);
 
-            string returnValue = output.EqualsOrdinal(originalPath.AsSpan())
+            string returnValue = !string.IsNullOrEmpty(originalPath) && output.EqualsOrdinal(originalPath.AsSpan())
```

Comment:
You do not need the check for Empty string. Maybe this can be just:

`string returnValue = ((originalPath != null) && output.EqualsOrdinal(originalPath)) ? originalPath : new string(output);`

Reply:
Sure, just habit. :)